### PR TITLE
Fixes #7899: Platform id continue to grow if highest id is not destroyed 

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/7899

### CODE REVIEW

File: rtf [hunk 1]
```diff
@@ -292,8 +297,19 @@ class Vagrant(Host):
...
+          next_id = max_pf_id + 1
+          # But we prefer to use min - 1
+          if min_pf_id > 1:
+            next_id = min_pf_id - 1
+          else:
+            # If min = 1, tries to find and id between two used ids
+            for x in range(1,max_pf_id):
+              if not x in ids:
+                next_id = x
+                break
```
Comment (peckpeck) [inline rtf]: if next_id is 256 or more, you should just fail

**Comment by peckpeck** [comment]
ping @VinceMacBuche

**Comment by VinceMacBuche** [comment]
PR updated

File: rtf [hunk 2]
```diff
@@ -292,8 +297,22 @@ class Vagrant(Host):
...
+          if next_id > 256:
```
Comment (peckpeck) [inline rtf]: >=

Comment (peckpeck) [inline rtf]: > or =

Comment (peckpeck) [inline rtf]: greater or equal

**Comment by VinceMacBuche** [comment]
PR updated

**Comment by peckpeck** [comment]
Good, merging

**Pull request merged**